### PR TITLE
ENH: Added the Final feature for all constants

### LINKED ISSUE #17507: ENH: Mark all constants as `Final`
A number of constants are present in the main numpy namespace, however their annotations 
are currently not marked as such.

TODO
------
- [ ] Mark constants (see below) with [`typing.Final`](https://docs.python.org/3/library/typing.html#typing.Final). For example, `Inf: float` should become `Inf: Final[float]`. 
Note that `Final` was introduced to the builtin `typing` module in python 3.8; for older versions it'll have to be imported from `typing_extensions`.

https://github.com/numpy/numpy/blob/29c58e632ce51e75e946014f076aba803ee57dde/numpy/__init__.pyi#L1600-L1641

### PR DESCRIPTION
**Pull request by roysti10**
Added the Final feature from typing to constants,
Closes #17507

### CODE REVIEW

File: numpy/__init__.pyi [hunk 1]
```diff
@@ -36,6 +36,7 @@ from typing import (
     Container,
     Callable,
     Dict,
+    Final,
```
Comment (BvB93) [inline numpy/__init__.pyi:39]: Can you move the `Final` import to the lines below?

`Final` was only added to the `typing` module in python 3.8; 
for earlier versions it requires a backport from `typing_extensions`.

https://github.com/numpy/numpy/blob/29c58e632ce51e75e946014f076aba803ee57dde/numpy/__init__.pyi#L60-L63

**Review by BvB93 — CHANGES_REQUESTED** [review]

File: numpy/__init__.pyi (hunk 1 above)
Comment (lucasace) [reply numpy/__init__.pyi]: Done!

File: numpy/__init__.pyi [hunk 2]
```diff
@@ -58,9 +58,9 @@ from typing import (
 )
 
 if sys.version_info >= (3, 8):
-    from typing import Literal, Protocol, SupportsIndex
+    from typing import Literal, Protocol, SupportsIndex,Final
```
Comment (BvB93) [inline numpy/__init__.pyi:61]:
```suggestion
    from typing import Literal, Protocol, SupportsIndex, Final
```

File: numpy/__init__.pyi [hunk 3]
```diff
@@ -1597,48 +1597,48 @@ def empty(
...
+NaN: Final[float]
+PINF: Final[float]
+PZERO: Final[float]
+e: Final[float]
+euler_gamma: Final[float]
+inf: Final[float]
+infty: Final[float]
+nan: Final[float]
+pi: Final[float]
+
```
Comment (BvB93) [inline numpy/__init__.pyi:1614]:
```suggestion

```
There seem to be a few empty spaces here.

**Review by BvB93 — COMMENTED** [review]
Looking good overall.
Two more small changes; afterwards it's good to go in my opinion.

**Review by BvB93 — APPROVED** [review]

**Comment by charris** [comment]
Thanks @lucasace .

**Pull request merged**